test/functional/forall/test-forall-atomic-utils.hpp

```diff
@@ -0,0 +1,80 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_FORALL_ATOMIC_UTILS_HPP__
+#define __TEST_FORALL_ATOMIC_UTILS_HPP__
+
+#include "RAJA/RAJA.hpp"
+
+#include "test-forall-utils.hpp"
+
+using AtomicSeqExecs = camp::list< RAJA::seq_exec >;
+
+using AtomicSeqPols = camp::list<
+#if defined(RAJA_TEST_EXHAUSTIVE)
+                                  RAJA::auto_atomic,
+                                  RAJA::builtin_atomic,
+#endif
+                                  RAJA::seq_atomic
+                                >;
+
+#if defined(RAJA_ENABLE_OPENMP)
+using AtomicOmpExecs = camp::list<
+#if defined(RAJA_TEST_EXHAUSTIVE)
+                                   RAJA::omp_for_nowait_exec,
+                                   RAJA::omp_parallel_for_exec,
+#endif
+                                   RAJA::omp_for_exec
+                                 >;
+
+using AtomicOmpPols = camp::list<
+#if defined(RAJA_TEST_EXHAUSTIVE)
+                                  RAJA::omp_atomic,
+                                  RAJA::builtin_atomic,
+#endif
+                                  RAJA::auto_atomic
+                                >;
+#endif  // RAJA_ENABLE_OPENMP
+
+#if defined(RAJA_ENABLE_CUDA)
+using AtomicCudaExecs = camp::list< RAJA::cuda_exec<256> >;
+
+using AtomicCudaPols = camp::list<
+#if defined(RAJA_TEST_EXHAUSTIVE)
+                                   RAJA::auto_atomic,
+#endif
+                                   RAJA::cuda_atomic
+                                 >;
+#endif  // RAJA_ENABLE_CUDA
+
+#if defined(RAJA_ENABLE_HIP)
+using AtomicHipExecs = camp::list< RAJA::hip_exec<256> >;
+
+using AtomicHipPols = camp::list<
+#if defined(RAJA_TEST_EXHAUSTIVE)
+                                   RAJA::auto_atomic,
+#endif
+                                   RAJA::hip_atomic
+                                >;
+#endif  // RAJA_ENABLE_HIP
+
+//
+// Atomic index types for segments
+//
+using AtomicTypeList = camp::list<
```

Comment:
Similar to comment above, I think we can just re-use the index type list in the forall utils file in the same directory instead of repeating it here.

Also, I really don't think there is a need to test float and double as index types.

Reply:
This is more problematic, they should be labeled as "atomic data types" rather than "index types". The atomics cannot work on any short's, and float/double are fairly essential atomic data types which ought to be included in testing. I think keeping this here (rather than complex ifdef'ing the other forall-utils header) is preferable.

Reply:
I see your point. I was confused about the labeling.

Reply:
Re-labeled these.